logstash-core/src/main/java/org/logstash/config/ir/graph/Vertex.java

```diff
@@ -49,11 +50,20 @@ public InvalidEdgeTypeException(String s) {
         }
     }
 
-    public Graph getGraph() {
+    @Override
+    public final int hashCode() {
+        return hashCode;
+    }
+
+    public final boolean equals(final Object other) {
+        return this == other;
```

Comment:
So, this is the default implementation of `Object.equals`. Is this just here to make that explicit?

Reply:
@andrewvc I wanted the `final` on the parent class here, to make sure that this is the case for all children as well (obviously it is, but it's annoying that I had to go through all of them when trying to track this down) => much easier to read this way.